dataclasses_jsonschema/__init__.py

```diff
@@ -416,6 +416,8 @@ def _decode_field(cls, field: str, field_type: Any, value: Any) -> Any:
         except (KeyError, TypeError):
             # Note: Only literal types composed of primitive values are currently supported
             if type(value) in JSON_ENCODABLE_TYPES and (field_type in JSON_ENCODABLE_TYPES or is_literal(field_type)):
+                if is_literal(field_type) and value not in get_args(field_type):
```

Comment:
I need to look into why the CI pipeline hasn't been run for this, but it looks like `get_args` was only added in python 3.8, so this will need importing from `typing_extensions` in lower python versions.